app/views/users/show.html.erb

```diff
@@ -0,0 +1,43 @@
+<p id="notice"><%= notice %></p>
+
+<p>
+  <strong>Full name:</strong>
+  <%= @user.full_name %>
+</p>
+
+<p>
+  <strong>Username:</strong>
+  <%= @user.username %>
+</p>
+
+<h2> List of <%= @user.username %> creates events: </h2>
+<ul>
+  <% @events.each do |eve| %>
+    <li> Title: <%= eve.title %> Location: <%= eve.location %> </li>
+  <% end %>
+</ul>
+
+<h2> List of events attended by <%= @user.username %> : </h2>
+<ul>
+    <% @attended_events.each do |event| %>
+      <li> Title: <%= event.title %> Location: <%= event.location %> Date: <%= event.event_date %> </li>
+    <% end %>
+</ul>
+
+
+<h2>List of Previous events </h2>
+<% @previous_events = @user.attended_events.where('event_date < ?', Time.now).to_a %>
+<% @upcomming_events = @user.attended_events.where('event_date > ?', Time.now).to_a %>
+<ul>
+  <% @previous_events.each do |event| %>
```

Comment:
**REQUIRED**
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.
Anything related to data processing should be placed in models passed to the views via the controllers.
**Resources.**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

Reply:
Thank you for the review. Issue solved.